docs/spec.openapi.yaml

```diff
@@ -0,0 +1,425 @@
+        - Authorization
+      description: Verifies the received token is still valid
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/authorization'
+      responses:
+        '200':
+          description: Account token is still valid
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/account'
+        '401':
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '500':
+          description: Internal error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+  /account/password:
+    patch:
+      tags:
+        - Account
+      description: Updates the username password
+      requestBody:
+        content:
+          application/json:
+            schema:
+              allOf:
+                - $ref: '#/components/schemas/authorization'
+                - type: object
+                  properties:
+                    currentPassword:
+                      type: string
+                    newPassword:
+                      type: string
+      responses:
+        '200':
+          description: Password updated successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '401':
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '500':
+          description: Internal error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+  /user/uuid/{username}:
+    get:
+      tags:
+        - Account
+      description: Obtain the uuid of an user by its username
+      parameters:
+        - in: path
+          name: username
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: User uuid
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/account'
+        '401':
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '500':
+          description: Internal error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+  /files:
+    post:
+      tags:
+        - Files
+      description: Create a new file
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/fileDetails'
```

Comment:
Check the [CreateFileReq.java](https://github.com/hawks-atlanta/gateway-java/blob/feat/soap-service-definition/app/src/main/java/gateway/soap/request/CreateFileReq.java) file. This endpoint should receive the `fileName`, the `location` (UUID of the parent directory) and the file itself, the `fileSize` is not important at this point.